c++/src/Writer.cc

```diff
@@ -338,6 +338,12 @@ namespace orc {
 
   const WriterId WriterImpl::writerId = WriterId::ORC_CPP_WRITER;
 
+  static void validateOptions(const WriterOptions& opts) {
```

Comment:
Is it better to check this inside WriterOptions::setCompressionBlockSize() ?

Reply:
Done, I've made it check when setting.